drivers/sensor/iis2iclx/iis2iclx.c

```diff
@@ -659,12 +657,15 @@ static int iis2iclx_init(const struct device *dev)
 
 #define IIS2ICLX_CONFIG_SPI(inst)					\
 	{								\
+		.ctx.read_reg = (stmdev_read_ptr) stmemsc_spi_read,	\
```

Comment:
This style technically works for C, but IMO it would be better as:

    .ctx = {
      .read_reg = ...,
      ...
    },

Though the same concern applies to `stmemsc_cfg`,where it would be more tedious.  So not blocking.

Reply:
Something like that would be a good compromise? Seems more readable in fact...                                                                                                        
                                                                                                        
```                                                                                                          
#define IIS2ICLX_CONFIG_SPI(inst)					\
	{								\
		.ctx = {						\
			.read_reg = (stmdev_read_ptr) stmemsc_spi_read,	\
			.write_reg = (stmdev_write_ptr) stmemsc_spi_write,	\
			.handle = (void *)&iis2iclx_config_##inst.stmemsc_cfg,	\
		},							\
		.stmemsc_cfg.spi = {					\
			.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),	\
			.spi_cfg =					\
				SPI_CONFIG_DT_INST(inst,		\
					   IIS2ICLX_SPI_OPERATION,	\
					   0),				\
		},							\
		.odr = DT_INST_PROP(inst, odr),				\
		.range = DT_INST_PROP(inst, range),			\
		COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, drdy_gpios),	\
			(IIS2ICLX_CFG_IRQ(inst)), ())			\
	}
```

Reply:
Yes, I like that better.